src/main/java/org/spongepowered/common/mixin/api/mcp/world/storage/WorldInfoMixin_API.java

```diff
@@ -160,9 +163,28 @@ public long getTotalTime() {
         return this.worldTime;
     }
 
-    @Intrinsic
-    public void worldproperties$setWorldTime(final long time) {
-        this.shadow$setWorldTime(time);
+    @Override
+    public void setWorldTime(final long time) {
+        final Optional<World> optionalTargetWorld = Sponge.getServer().getWorld(getUniqueId());
```

Comment:
There are two things that I'm a little iffy about here - one should be easy to fix, the other, not so much (a similar issue exists with your previous PR):

* The event construction and firing should be wrapped in `ShouldFire.CHANGE_WORLD_TIME` (I think?), as this is going to fire 20 times a second _at least_. Search for `ShouldFire` in your IDE and see how we do it with other big hitting events - this should avoid this event being called if no-one is listening for it.
* This returns `void`. A caller will set the time but there is no way to know what happened with that request - we've had this before with cancellable events. This directly feeds into your question about the multi-world command. Thus, I'm a little iffy on this... annoyingly because this would be a nice PR to accept.

For your multi world command stuff, you could just check what the world time is as it's been set at this point - and if it's the same as before, you know nothing changed. But my concern is that plugins won't know about this and could cause weird behaviour.

Then again, this might be an edge case and users who install plugins that do this will be fully aware of the consequences. If this is the case, we _must_ update the Javadocs.

There is also the option of introducing a new API method, but naming it would be annoying as `setWorldTime` is a good name already. Could be `long setTime(long)`, of course...

Consider how this would fit into API 8. In particular, I introduced `MinecraftDayTime` - this is going to change anyway so it would be great if you could at least specify the method changes you think you'll need so we don't get caught unawares later!